joeynmt/encoders.py

```diff
@@ -142,6 +141,109 @@ def forward(self, embed_src: Tensor, src_length: Tensor, mask: Tensor) \
         # final: batch x directions*hidden
         return output, hidden_concat
 
-    @property
     def __repr__(self):
         return "%s(%r)" % (self.__class__.__name__, self.rnn)
+
+
+class TransformerEncoder(Encoder):
+    """
+    Transformer Encoder
+    """
+
+    #pylint: disable=unused-argument
+    def __init__(self,
+                 hidden_size: int = 512,
+                 ff_size: int = 2048,
+                 num_layers: int = 8,
+                 num_heads: int = 4,
+                 dropout: float = 0.1,
+                 freeze: bool = False,
+                 **kwargs):
+        """
+        Initializes the Transformer.
+        :param hidden_size: hidden size and size of embeddings
```

Comment:
Do they have to be the same? Can we disentangle them?

Reply:
They have to be.